### PR TITLE
Included members would remain in a list after data source was removed

### PR DESCRIPTION
**Pull request by salaun-urennes1**
If an include_xx parameter was removed, sync_include() was not triggered. If no other include_xx remained, included members would remain in the list with no way to cleanup list members because sync_include() cannot be run on a list with no include_xx parameter.

### CODE REVIEW

**Comment by ikedas** [comment]
I don't agree to this change.  I think it is natural that, when a data source is removed, the users included only by that data source  will be removed.

**Comment by salaun-urennes1** [comment]
Hi Soji, so I guess you agree with the PR?
However I don't think the "enhancement" tag is appropriate. I'd rather tag this PR as bug fix.

File: src/lib/Sympa/List.pm [hunk 1]
```diff
@@ -4165,10 +4165,6 @@ sub sync_include {
 
     $role ||= 'member';    # Compat.<=6.2.54
 
-    return 0
-        unless $self->has_data_sources($role)
-        or $self->has_included_users($role);
-
```
Comment (ikedas) [inline src/lib/Sympa/List.pm:4171]: I think this paragraph means that: Updating inclusion will be performed when:
- The list has any data sources, or
- The list has users who have been included, even if the list already doesn't have any data sources.

So it need not be removed for me.

File: src/lib/Sympa/Request/Handler/include.pm [hunk 2]
```diff
@@ -131,10 +131,6 @@ sub _twist {
     die 'bug in logic. Ask developer'
         unless $role and grep { $role eq $_ } qw(member owner editor);
 
-    return 0
-        unless $list->has_data_sources($role)
-        or $list->has_included_users($role);
-
```
Comment (ikedas) [inline src/lib/Sympa/Request/Handler/include.pm:137]: Maybe the same as previous comment.

File: src/cgi/wwsympa.fcgi.in [hunk 3]
```diff
@@ -10613,7 +10613,7 @@ sub do_edit_list {
     }
 
     my $data_source_updated_member = 1
-        if grep { $config->get_change($_) }
+		if grep { exists $config->{_changes}{$_} }
```
Comment (ikedas) [inline src/cgi/wwsympa.fcgi.in:10616]: What is the motivation of this change?

Comment (salaun-urennes1) [reply src/cgi/wwsympa.fcgi.in:10616]: I noticed the previous code did not work. I could not understand why it failed and found this bypass to make it work. Sorry, I should have added a comment to explain his in the PR.

**Review by ikedas — COMMENTED** [review]
I think this pr doesn't contain significant changes and I suggest withdrawal.

**Comment by salaun-urennes1** [comment]
I don't agree with you proposition since the current behavior is not consistent.
If the proposed PR does not meet your expectations, feel free to propose another alternative to meet the expected bahavior.

**Comment by ikedas** [comment]
> I don't agree with you proposition since the current behavior is not consistent.
> If the proposed PR does not meet your expectations, feel free to propose another alternative to meet the expected bahavior.

Please explain how your modifications (including three stanzas) will change the behavior.

**Comment by ikedas** [comment]
Looks like a duplicate of #858 that has been fixed by #905.  I close this for now.

@salaun-urennes1 , if you have complain, please submit the new [bug report](https://github.com/sympa-community/sympa/issues/new?template=bug-report.md), and then submit pr if possible.

**Comment by salaun-urennes1** [comment]
I confirm that problem is fixed with Sympa 6.2.68 :-)
